src/ninshiki_cpp_detector.cpp

```diff
@@ -18,36 +18,39 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-#include "ninshiki_cpp/ninshiki_cpp.hpp"
-#include "rclcpp/rclcpp.hpp"
-
 #include <memory>
 #include <string>
 
+#include "ninshiki_cpp/ninshiki_cpp.hpp"
+#include "rclcpp/rclcpp.hpp"
+
 int main(int argc, char ** argv)
 {
   rclcpp::init(argc, argv);
-  shisen_cpp::Options options;
 
   // Default Value
   std::string path = "";
   std::string topic_name = "";
-  std::string detection_method = "yolo";
+
   int gpu = 0;
   int myriad = 0;
   int frequency = 96;
 
+  std::shared_ptr<ninshiki_cpp::detector::DnnDetector>dnn_detector = nullptr;
+  std::shared_ptr<ninshiki_cpp::detector::ColorDetector>color_detector = nullptr;
+  std::shared_ptr<ninshiki_cpp::detector::LBPDetector>lbp_detector = nullptr;
```

Comment:
All of them are already derived from the `ninshiki_cpp::detector::Detector` class, it would be better to use that instead for all of them:
```suggestion
  std::shared_ptr<ninshiki_cpp::detector::Detector> detector = nullptr;
```

If you're stuck with handling the `DnnDetector::set_computation_method` or `ColorDetector::load_configuration` method, then simply just put that in the base class and make the implementation for the other class to do nothing.

Reply:
the program should allow more than one detection method. Should I use vector of ```std::shared_ptr<ninshiki_cpp::detector::Detector> detector``` with fixed size 3 instead?

Reply:
Aah i just realized what this program trying to do with the detectors, if it ended up going to be passed separately to the ninshiki_cpp::node::NinshikiCppNode` then just keep the code like this for now. Forget my previous suggestion.